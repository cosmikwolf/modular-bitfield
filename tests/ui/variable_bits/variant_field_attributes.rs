use modular_bitfield::prelude::*;

// Test: Missing variant_discriminator field
#[bitfield(variable_bits = (32, 64))]
pub struct MissingDiscriminator {
    #[variant_data]
    data: B32,
}

// Test: Missing variant_data field  
#[bitfield(variable_bits = (32, 64))]
pub struct MissingData {
    #[variant_discriminator]
    discriminator: B4,
}

// Test: Multiple variant_discriminator fields
#[bitfield(variable_bits = (32, 64))]
pub struct MultipleDiscriminators {
    #[variant_discriminator]
    disc1: B4,
    #[variant_discriminator]
    disc2: B4,
    #[variant_data]
    data: B24,
}

// Test: Multiple variant_data fields
#[bitfield(variable_bits = (32, 64))]
pub struct MultipleData {
    #[variant_discriminator]
    discriminator: B4,
    #[variant_data]
    data1: B20,
    #[variant_data]
    data2: B8,
}

// Test: Invalid discriminator attribute format
#[bitfield]
pub struct InvalidDiscriminatorFormat {
    #[variant_discriminator = "value"]
    field1: B4,
    field2: B28,
}

// Test: Invalid data attribute format
#[bitfield]
pub struct InvalidDataFormat {
    field1: B4,
    #[variant_data(param)]
    field2: B28,
}

// Test: Using variant attributes without variable_bits
#[bitfield]
pub struct NoVariableBits {
    #[variant_discriminator]
    discriminator: B4,
    #[variant_data]
    data: B28,
}

fn main() {}