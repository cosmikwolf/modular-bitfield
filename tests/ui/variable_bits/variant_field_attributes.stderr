error: encountered unsupported #[bitfield] attribute
 --> tests/ui/variable_bits/variant_field_attributes.rs:4:12
  |
4 | #[bitfield(variable_bits = (32, 64))]
  |            ^^^^^^^^^^^^^^^^^^^^^^^^

error: encountered unsupported #[bitfield] attribute
  --> tests/ui/variable_bits/variant_field_attributes.rs:11:12
   |
11 | #[bitfield(variable_bits = (32, 64))]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^

error: encountered unsupported #[bitfield] attribute
  --> tests/ui/variable_bits/variant_field_attributes.rs:18:12
   |
18 | #[bitfield(variable_bits = (32, 64))]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^

error: encountered unsupported #[bitfield] attribute
  --> tests/ui/variable_bits/variant_field_attributes.rs:29:12
   |
29 | #[bitfield(variable_bits = (32, 64))]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^

error: encountered invalid format for #[variant_discriminator] field attribute, expected #[variant_discriminator]
  --> tests/ui/variable_bits/variant_field_attributes.rs:42:5
   |
42 |     #[variant_discriminator = "value"]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: encountered invalid format for #[variant_data] field attribute, expected #[variant_data]
  --> tests/ui/variable_bits/variant_field_attributes.rs:51:5
   |
51 |     #[variant_data(param)]
   |     ^^^^^^^^^^^^^^^^^^^^^^
